docs/advanced-topics/hybrid-models.md

```diff
@@ -0,0 +1,137 @@
+
+{% hint style="warning" %}
+If model IP protection is important, care must be taken in choosing the parts of a model to be executed on the cloud. Some
+black-box model stealing attacks rely on knowledge distillation
+or on differential methods. As a general rule, the difficulty
+to steal a machine learning model is proportional to the size of the model, in terms of numbers of parameters and model depth.
+{% endhint %}
+
+The hybrid model deployment API provides an easy way to integrate the [standard deployment procedure](client_server.md) into neural network style models that are compiled with [`compile_brevitas_qat_model`](../developer-guide/api/concrete.ml.torch.compile.md#function-compile_brevitas_qat_model) or [`compile_torch_model`](../developer-guide/api/concrete.ml.torch.compile.md#function-compile_torch_model).
+
+## Compilation
+
+To use hybrid model deployment, the first step is to define what part of the PyTorch neural network model must be executed in FHE. The model part must be a `nn.Module` and is identified by its key in the original model's `.named_modules()`.
+
+```python
+import numpy as np
+import os
+import torch
+
+from pathlib import Path
+from torch import nn
+
+from concrete.ml.torch.hybrid_model import HybridFHEModel, tuple_to_underscore_str
+from concrete.ml.deployment import FHEModelServer
+
+
+class FCSmall(nn.Module):
+    """Torch model for the tests."""
+
+    def __init__(self, dim):
+        super().__init__()
+        self.seq = nn.Sequential(nn.Linear(dim, dim), nn.ReLU(), nn.Linear(dim, dim))
+
+    def forward(self, x):
+        return self.seq(x)
+
+model = FCSmall(10)
+model_name = "FCSmall"
+submodule_name = "seq.0"
+
+inputs = torch.Tensor(np.random.uniform(size=(10, 10)))
+# Prints ['', 'seq', 'seq.0', 'seq.1', 'seq.2']
+print([k for (k, _) in model.named_modules()])
+
+# Create a hybrid model
+hybrid_model = HybridFHEModel(model, [submodule_name])
+hybrid_model.compile_model(
+    inputs,
+    n_bits=8,
+)
+
+
+models_dir = Path(os.path.abspath('')) / "compiled_models"
+models_dir.mkdir(exist_ok=True)
+model_dir = models_dir / model_name
+
+```
+
+<!--pytest-codeblocks:skip-->
+
+```python
+hybrid_model.save_and_clear_private_info(model_dir, via_mlir=True)
+```
+
+## Server Side Deployment
+
+The [`save_and_clear_private_info`](../developer-guide/api/concrete.ml.torch.hybrid_model.md#method-save_and_clear_private_info) function serializes the FHE circuits
+corresponding to the various parts of the model that were chosen to be moved
+server-side. Furthermore it saves all necessary information required
+to serve these sub-models with FHE, using the [`FHEModelDev`](../developer-guide/api/concrete.ml.deployment.fhe_client_server.md#class-fhemodeldev) class.
```

Comment:
Also something important is that the save_and_clear_private_info removed all the IP related weights.

Reply:
ok, I'll add that